### PR TITLE
Fix windows encoding error and upgrade extensions

### LINKED ISSUE #51: py3.6 encoding issue with negative numbers
I've found an issue across both linux and windows in py3.6 (and I believe 3.7) here when "minus signs" are used - at least in the axis ticks (\u2212)

https://github.com/FlyRanch/figurefirst/blob/abc43a21bc1a61ba0ef509628aa759e0d73495b3/figurefirst/svg_to_axes.py#L1244

``` python
Traceback (most recent call last):
  File "C:\Users\aaron\Miniconda3\envs\test\lib\site-packages\figurefirst\svg_to_axes.py", line 1235, in write_svg
    self.output_xml.writexml(outfile, encoding='utf-8')
  File "C:\Users\aaron\Miniconda3\envs\test\lib\xml\dom\minidom.py", line 1796, in writexml
    node.writexml(writer, indent, addindent, newl)
  File "C:\Users\aaron\Miniconda3\envs\test\lib\xml\dom\minidom.py", line 870, in writexml
    node.writexml(writer, indent+addindent, addindent, newl)
  File "C:\Users\aaron\Miniconda3\envs\test\lib\xml\dom\minidom.py", line 870, in writexml
    node.writexml(writer, indent+addindent, addindent, newl)
  File "C:\Users\aaron\Miniconda3\envs\test\lib\xml\dom\minidom.py", line 870, in writexml
    node.writexml(writer, indent+addindent, addindent, newl)
  [Previous line repeated 4 more times]
  File "C:\Users\aaron\Miniconda3\envs\test\lib\xml\dom\minidom.py", line 1185, in writexml
    writer.write("%s<!--%s-->%s" % (indent, self.data, newl))
  File "C:\Users\aaron\Miniconda3\envs\test\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2212' in position 5: character maps to <undefined>
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "<input>", line 60, in <module>
  File "C:\Users\aaron\Miniconda3\envs\test\lib\site-packages\figurefirst\svg_to_axes.py", line 1239, in write_svg
    outfile.write(self.output_xml.toxml().encode('ascii', 'xmlcharrefreplace'))
TypeError: write() argument must be str, not bytes
```

I've been fixing it by replacing 

```python
outfile.write(self.output_xml.toxml().encode('ascii', 'xmlcharrefreplace'))
```

with

```python
outfile.write(self.output_xml.toxml().encode('utf-8', 'xmlcharrefreplace'))
```

And it's not caused any issues

I've attached a minimal affected eps 

[ex.zip](https://github.com/FlyRanch/figurefirst/files/3640157/ex.zip)

And a minimal affected code

```python
import figurefirst as fifi

layout = fifi.svg_to_axes.FigureLayout('ex.svg')
layout.make_mplfigures()
layout.fig.set_facecolor('None')
ex = layout.axes['ex']
ex.plot([1, 2], [3, 4])
fifi.mpl_functions.adjust_spines(ex, spines='left', yticks=[-1, -2])
layout.insert_figures('panels', cleartarget=True)
layout.write_svg('ex.svg')

```

### LINKED ISSUE #52: Inkscape 1 support
Hi, there are some changes with Inkscape 1.

I know @kukasyachka has some fixes at https://github.com/kukasyachka/figurefirst/tree/inkscape1, but so far this is not a complete set of the updates needed.

### PR DESCRIPTION
**Pull request by psilentp**
Closes #51 and closes #52 

* Fixes an issue with utf-8 encoding on windows. Error raised on windows runner in a4e4be8.
* Upgrades extensions and extension install script to optionally install extensions for inkscape >= 1.0.0 
* Applies black formatting in a number of places
* Adds CI to run tests on linux and windows.

### CODE REVIEW

**Comment by psilentp** [comment]
Dev notes (because I don't have a better place to put them):
I was able to raise the error described in #51 running the `python:windowservercore` docker image on a windows virtual box using vagrant.

Specifically I followed the instructions at: https://github.com/StefanScherer/windows-docker-machine#mounting-volumes-from-your-windows-machine to fire up a windows virtual box server.

```
$ git clone https://github.com/StefanScherer/windows-docker-machine
$ cd windows-docker-machine
$ vagrant up --provider virtualbox 2019-box
```
Then pointed my docker context to the windows server.

`$ docker context use 2019-box`

I was then able to shell into the official python windows image and mounted `figurefirst` by running this command from the `figurefirst` directory on the host: 

`$ docker run -it -v C:$(pwd):C:$(pwd) python:windowsservercore powershell`

In the powershell I found `figurefirst` at `C:\Users\host_username\path\on\host\figurefirst\`

I installed `figurefirst` with `pip install -e .` and ran the `example_negative_labels.py` script to raise the error. 

Note: I was not able to run all the tests because of a numpy import error. It seems that I am unable to load the c-extensions in the (double) virtualized environment. It might be worth seeing if we can run the test as part of a CI pipline using github actions (which appears to have a windows runner).

**Comment by aaronshifman** [comment]
@psilentp I can confirm that the updates pass all tests as well as the failing example in #51 on Windows. There was another issue brought about by the change by someone else. I've asked him if this resolves it.

**Comment by gazzar** [comment]
I can confirm this fixes the utf issue for me too on a native Windows machine. The utf problem was also occurring when I added Greek characters to the svg template but that works now; thanks!

File: figurefirst_scripts/install_inkscape_ext.py [hunk 1]
```diff
@@ -74,39 +76,62 @@ def copy_extensions(src_dir, tgt_dir, overwrite_all=False, overwrite_none=False)
...
         "~/Library/Application\ Support/org.inkscape.Inkscape/config/inkscape/extensions ; "
         "On Windows, defaults to "
-        "C:\\Program Files\\Inkscape\\share\\extensions"
+        "C:\\Program Files\\Inkscape\\share\\extensions",
+    )
+    parser.add_argument(
+        "--inkscape_major_version",
+        default=0,
+        help="Set this to the major version of inkscape you will be using e.g. 1 for inkscape >=1.0",
+    )
```
Comment (clbarnes) [inline figurefirst_scripts/install_inkscape_ext.py:101]: I think we could default to 1 here, as it's the latest version and may even be stable.

**Comment by clbarnes** [comment]
Great work! I hadn't even noticed inkscape was at 1.0. And awesome to have the tests such as they are running under windows too.

I am personally very happy with using black, and would be in favour of adding it (and possibly some linting) to CI, but that's a separate job.

File: figurefirst_scripts/install_inkscape_ext.py (hunk 1 above)
Comment (psilentp) [reply figurefirst_scripts/install_inkscape_ext.py:101]: Good to get that feedback - that was my original instinct, but I haven't tested the updated extensions a ton. I think I'll do a little more manual testing on the extensions to make sure they work as expected and then change the default as you suggest (maybe with a callout in the readme).

**Comment by psilentp** [comment]
🤦 I missed that @kukasyachka had already tested the extensions. Setting the default to 1.0 and merging.

**Pull request merged**